rcode-reader/src/main/java/eu/rssw/pct/RCodeInfo.java

```diff
@@ -0,0 +1,723 @@
+      for (Parameter p : parameters) {
+        if (sb2.length() > 0) {
+          sb2.append(", ");
+        }
+        sb2.append(p.toString());
+      }
+      sb.append(sb2.toString()).append(')');
+
+      return sb.toString();
+    }
+  }
+
+  public enum FunctionType {
+    MAIN,
+    CONSTRUCTOR,
+    METHOD,
+    FUNCTION,
+    PROCEDURE,
+    EXTERNAL_PROCEDURE,
+    DLL_PROCEDURE,
+    DESTRUCTOR;
+
+    public static FunctionType getTypeFromString(String str) {
+      switch (str.toUpperCase()) {
+        case "CONST": return CONSTRUCTOR;
+        case "MAIN": return MAIN;
+        case "METH": return METHOD;
+        case "FUNC": return FUNCTION;
+        case "PROC": return PROCEDURE;
+        case "EXT": return EXTERNAL_PROCEDURE;
+        case "DLL": return DLL_PROCEDURE;
+        case "DEST": return DESTRUCTOR;
+      }
+
+      return null;
+    }
+  }
+
+  public enum SigAccessType {
+    PUBLIC,
+    PRIVATE,
+    PROTECTED,
+    STATIC,
+    ABSTRACT,
+    FINAL,
+    OVERRIDE;
+
+    public static EnumSet<SigAccessType> getTypeFromString(String str) {
+      int val = Integer.valueOf(str);
+      EnumSet<SigAccessType> set = EnumSet.noneOf(SigAccessType.class);
+      switch (val & 0x07) {
+        case 1:
+          set.add(PUBLIC);
+          break;
+        case 2:
+          set.add(PRIVATE);
+          break;
+        case 4:
+          set.add(PROTECTED);
+          break;
+      }
+      if ((val & 0x08) != 0)
+        set.add(STATIC);
+      if ((val & 0x10) != 0)
+        set.add(ABSTRACT);
+      if ((val & 0x20) != 0)
+        set.add(FINAL);
+      if ((val & 0x40) != 0)
+        set.add(OVERRIDE);
+
+      return set;
+    }
+  }
+
+  public enum ParameterType {
+    INPUT,
+    OUTPUT,
+    INPUT_OUTPUT,
+    BUFFER;
+
+    public static ParameterType getTypeFromString(String str) {
+      switch (str) {
+        case "1":
+          return INPUT;
+        case "2":
+          return OUTPUT;
+        case "3":
+          return INPUT_OUTPUT;
+        case "4":
+          return BUFFER;
+        default:
+          return null;
+      }
+    }
+  }
+
+  public static class Parameter {
+    private ParameterType type;
+    private String name;
+    private DataType datatype;
+    private int extent;
+    private String classType;
+
+    @Override
+    public String toString() {
+      StringBuilder sb = new StringBuilder();
+      if (type != null) {
+        sb.append(type.toString());
+      }
+      sb.append(" PARAMETER ");
+      if (type == ParameterType.BUFFER) {
+        sb.append(" FOR ").append(name);
+      } else {
+        sb.append(name).append(" AS ").append(datatype);
+        if (datatype == DataType.CLASS) {
+          sb.append('{').append(classType).append('}');
+        }
+        if (extent > 0) {
+          sb.append(" EXTENT ").append(extent);
+        }
+      }
+
+      return sb.toString();
+    }
+  }
+
+  public static class TempTable {
+    private String name;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "name" private field. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS1068)